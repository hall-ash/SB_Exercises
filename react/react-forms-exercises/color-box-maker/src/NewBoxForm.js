import React, {useState} from 'react';

const NewBoxForm = ({addBox}) => {

  const INITIAL_STATE = {
    bgColor: "",
    width: "",
    height: "",
  };

  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = evt => {
    const { name, value } = evt.target;
    setFormData(fData => ({
      ...fData,
      [name]: value,
    }));
  };

  const handleSubmit = evt => {
    evt.preventDefault();
    addBox(formData);
    setFormData(INITIAL_STATE);
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="bgColor">Background color:</label>
      <input type="text" id="bgColor" name="bgColor" onChange={handleChange} value={formData.bgColor}/><br></br>
      <label htmlFor="width">Width:</label>
      <input type="text" id="width" name="width" onChange={handleChange} value={formData.width}/><br></br>
      <label htmlFor="height">Height:</label>
      <input type="text" id="height" name="height" onChange={handleChange} value={formData.height}/><br></br>
      <button>Add</button>
    </form>
  );

};

export default NewBoxForm;
