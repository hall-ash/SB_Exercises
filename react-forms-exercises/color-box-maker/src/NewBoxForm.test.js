import React from 'react';
import { render } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

test('renders without crashing', () => {
  render(<NewBoxForm />);
});

test('it matches snapshot', () => {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});