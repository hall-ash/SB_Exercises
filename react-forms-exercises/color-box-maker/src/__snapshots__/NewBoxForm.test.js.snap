// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="bgColor"
    >
      Background color:
    </label>
    <input
      id="bgColor"
      name="bgColor"
      type="text"
      value=""
    />
    <br />
    <label
      for="width"
    >
      Width:
    </label>
    <input
      id="width"
      name="width"
      type="text"
      value=""
    />
    <br />
    <label
      for="height"
    >
      Height:
    </label>
    <input
      id="height"
      name="height"
      type="text"
      value=""
    />
    <br />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
